lib/Dialect/TT/IR/TTOpsTypes.cpp

```diff
@@ -479,6 +502,17 @@ uint64_t TileType::getSizeBytes() const {
   }
 }
 
+SystemDescAttr mlir::tt::getCurrentScopeSystemDesc(mlir::Operation *op) {
+  while (op) {
+    if (auto systemDesc =
```

Comment:
Each OP can now have system desc attr? Or only module and funcops? Can system desc be different between different func ops?

Reply:
Only ModuleOps can have a system desc, this is just a convenience method that gets the system desc of the current scope. This interface method forwards to `getCurrentScopeSystemDesc` which walks up parent ops until it hits the module op.

Reply:
It allows you to get at the system desc from anywhere. If you're in some rewrite pattern you don't necessarily have a direct handle to the enclosing module.

Reply:
isa<ModuleOp>  check might be better than? You clearly state what type has the attr and you will throw if not found where expected instead of returning nullptr, or there is a valid case for nullptr to be returned?

Reply:
Yeah I copy pasted this from `getCurrentScopeDevice` and I think in the future the device attribute could appear on specialized ops (like `ttir.with_device` op which reinterprets the device grid), but it makes more sense to be more strict here.  I'll fix this with your suggestion.